app/controllers/provider/passwords_controller.rb

```diff
@@ -28,15 +30,15 @@ def destroy
   end
 
   def show
-    if params[:password_reset_token].present? && session[:password_reset_token].blank?
+    if params.require(:password_reset_token).present? && session[:password_reset_token].blank?
       new_token = @user.generate_lost_password_token
       session[:password_reset_token] = new_token
       redirect_to provider_password_path
     end
   end
 
   def update
-    user = params[:user]
+    user = params.require(:user)
```

Comment:
You would need to use permit correctly here

```suggestion
    user = params.permit(user: %i[password password_confirmation])
```